apps/ejabberd/src/mod_caps.erl

```diff
@@ -575,22 +575,22 @@ concat_info(Els) ->
                              Els)).
 
 concat_xdata_fields(Fields) ->
```

Comment:
We try to do three things in the function:
- Extract fields;
- Concat fields;
- Get FORM_TYPE field.
It would MUCH cleaner code if it was split into functions doing one and only one thing.
Yes, it would be less performant, but much more easier to read.